libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m

```diff
@@ -859,11 +858,19 @@ - (void)loadVFPingPage
 - (void)refreshCredentials:(SFOAuthCredentials *)credentials
                  completion:(SFOAuthFlowSuccessCallbackBlock)completionBlock
                   failure:(SFOAuthFlowFailureCallbackBlock)failureBlock {
-    if ([SFUserAccountManager sharedInstance].useLegacyAuthenticationManager) {
-        [[SFAuthenticationManager sharedManager] refreshCredentials:credentials completion:completionBlock failure:failureBlock];
-    } else {
-        [[SFUserAccountManager sharedInstance] refreshCredentials:credentials completion:completionBlock failure:failureBlock];
```

Comment:
Is this something that needs to be fixed further up the chain?  Seems like `[SFUserAccountManager refreshCredentials:completion:failure:]` shouldn't be allowed to initiate a User Agent flow....

Reply:
It doesn't trigger a user agent flow per se, it triggers a refresh flow. But the refresh flow is still using `SFOAuthCoordinator` and spinning up `SFSDKOAuthClient`, etc., instead of using `SFOAuthSessionRefresher`. I do want to refactor that code and make enhancements there, but it's too close to `6.1` for that. The goal of this exercise is to have the same design on `iOS` and `Android` as much as possible.

Reply:
Got it, thanks.  I don't see anything wrong with the specifics of your implementation.  Just was curious on the upstream question.

Reply:
More meat for my `7.0 Enhancements` doc ;-)